src/AzureAppConfigurationImpl.ts

```diff
@@ -91,10 +91,39 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         this.#adapters.push(new JsonKeyValueAdapter());
     }
 
+    // ReadonlyMap APIs
     get<T>(key: string): T | undefined {
         return this.#configMap.get(key);
     }
 
+    forEach(callbackfn: (value: any, key: string, map: ReadonlyMap<string, any>) => void, thisArg?: any): void {
+        this.#configMap.forEach(callbackfn, thisArg);
+    }
+
+    has(key: string): boolean {
+        return this.#configMap.has(key);
+    }
+
+    get size(): number {
+        return this.#configMap.size;
+    }
+
+    entries(): IterableIterator<[string, any]> {

```

Comment:
I was looking at [this documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/extends#avoiding_inheritance), and they are defining some of these methods as generator functions. I'm guessing that's not useful in our scenario because we don't expect any async operations here?

Reply:
In short, not related to sync or async. The impl of APIs are designed to return exactly the same as a Map does. BTW for Map, all above APIs are sync operations except `forEach`. 

The MDN implementation is just a demonstration of pros not using inheritance in some cases. The `ReadOnlyMap` impl mentioned is not standard but provides correct results via expected usage. It's like we create ReadOnlyMap.keys as a generator, in terms of what to yield, it delegates to the Map.keys() iterator. But the type is different from Map.keys. 
Below comparison should be clear enough.
```console
// paste ReadOnlyMap impl from MDN
> a = new Map()
Map(0) {}
> a.set("k", "v")
Map(1) { 'k' => 'v' }
> ak=a.keys()
[Map Iterator] { 'k' }
> ak.next()
{ value: 'k', done: false }
> ak.next()
{ value: undefined, done: true }

> b = new ReadOnlyMap([["k", "v"]])
ReadOnlyMap {}
> bk = b.keys()
Object [Generator] {}   // <--- different type and toString() impl
> bk.next()
{ value: 'k', done: false }  // <--- correctly yield 'k'
> bk.next()
{ value: undefined, done: true }
```

My ReadonlyMap implementation here doesn't directly get generator involved. It simply returns exactly the same `IterableIterator` type directly.

If you are interested in the concepts, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators for reference about iterator/generator/iterable...

As for the async usage you mentioned, that's something like AsyncIterableIterator, which implements a different symbol [Symbol.asyncIterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/asyncIterator) rather than [Symbol.iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/iterator)